File: backend/src/main/java/com/ai_kids_care/v1/entity/AppreciationLetter.java
package com.ai_kids_care.v1.entity;

import com.ai_kids_care.v1.type.AppreciationTargetTypeEnum;
import com.ai_kids_care.v1.type.StatusEnum;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;

@Getter
@Setter
@Entity
@Table(name = "appreciation_letters")
public class AppreciationLetter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "letter_id", nullable = false)
    private Long id;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "kindergarten_id", nullable = false)
    private Kindergarten kindergarten;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "sender_user_id", nullable = false)
    private User senderUser;

    @NotNull
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "target_type", columnDefinition = "appreciation_target_type_enum")
    private AppreciationTargetTypeEnum targetType;

    @NotNull
    @Column(name = "target_id", nullable = false)
    private Long targetId;

    @NotNull
    @Column(name = "title", nullable = false, length = Integer.MAX_VALUE)
    private String title;

    @NotNull
    @Column(name = "content", nullable = false, length = Integer.MAX_VALUE)
    private String content;

    @NotNull
    @Column(name = "is_public", nullable = false)
    private Boolean isPublic;

    @NotNull
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "status", columnDefinition = "status_enum")
    private StatusEnum status;

    @CreationTimestamp
    @ColumnDefault("now()")
    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @ColumnDefault("now()")
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;


}

File: frontend/src/types/appreciationLetter.ts
/**
 * DB `appreciation_target_type_enum` / 백엔드 `AppreciationTargetTypeEnum`
 */
export type AppreciationTargetType = 'KINDERGARTEN' | 'TEACHER';

/** 백엔드 `AppreciationLetterVO` (JSON camelCase) */
export type AppreciationLetterVO = {
  letterId: number;
  senderName: string | null;
  targetType: string;
  targetName: string | null;
  title: string;
  content: string;
  isPublic: boolean;
  editable: boolean;
  createdAt: string;
  updatedAt: string;
};

export const APPRECIATION_TARGET_TYPE_OPTIONS: { value: AppreciationTargetType; label: string }[] = [
  { value: 'KINDERGARTEN', label: '유치원' },
  { value: 'TEACHER', label: '교사' },
];
